Client/Frontend/Toolbar+URLBar/ReaderModeButton.swift

```diff
@@ -0,0 +1,63 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+class ReaderModeButton: UIButton {
+
+    // MARK: - Initializers
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        adjustsImageWhenHighlighted = false
+        setImage(UIImage.templateImageNamed("reader"), for: .normal)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Variables
+    
+    var selectedTintColor: UIColor?
+    var unselectedTintColor: UIColor?
+    override var isSelected: Bool {
+        didSet {
+            self.tintColor = (isHighlighted || isSelected) ? selectedTintColor : unselectedTintColor
+        }
+    }
+
+    override open var isHighlighted: Bool {
+        didSet {
+            self.tintColor = (isHighlighted || isSelected) ? selectedTintColor : unselectedTintColor
+        }
+    }
+
+    override var tintColor: UIColor! {
+        didSet {
+            self.imageView?.tintColor = self.tintColor
+        }
+    }
+
+    var savedReaderModeState = ReaderModeState.unavailable
```

Comment:
```suggestion
    private var savedReaderModeState = ReaderModeState.unavailable
```

I checked in Xcode and it seems like nothing else is using it. Builds fine for me. Same for the other suggestion here that was missed. :)